src/ApiCall.js

```diff
@@ -0,0 +1,254 @@
+// ApiCall.js
+import axios from 'axios';
+import { useState, useEffect } from 'react';
+import GalleryItem from './GalleryItem';
+import Recipe from './Recipe';
+import ErrorMessage from './ErrorMessage';
+import firebase from './firebase';
+import { getDatabase, ref, onValue, push, remove } from 'firebase/database';
+
+export default function ApiCall(props) {
+
+    const [recipes, setRecipes] = useState([]);
+    
+    // toggle between searched recipe and each recipe
+    const [showRecipeCard, setShowRecipeCard] = useState(false);
+
+    const [showSaved, setShowSaved] = useState(false)
+    
+    // show more button
+    const [showMore, setShowMore] = useState(null);
+    
+    // toggle between false and true class styling
+    const [error, setError] = useState(false);
+    
+    // get the current recipe user clicked
+    const [display, setDisplay] = useState(null);
+
+    // state for saved recipes
+    const [savedRecipes, setSavedRecipes] = useState([]);
+
+    // api call
+    useEffect( () => {
+        setTimeout( () => {
+            axios({
+                url: 'https://api.edamam.com/api/recipes/v2',
```

Comment:
if api call takes some time to back maybe try an asynchronous call if not it looks great otherwise